src/components/PhotoUpload/PhotoUploadCaptchaToolbar.js

```diff
@@ -0,0 +1,381 @@
+    top: 0;
+    height: 50px;
+    min-height: 50px;
+    z-index: 1;
+    border-bottom: 1px ${colors.borderColor} solid;
+    background: white;
+
+    > input {
+      font-size: 1em;
+      flex-grow: 1;
+      margin: 6px 20px 6px 0;
+      border: none;
+      border-radius: 0;
+      border-bottom: 1px solid ${colors.linkColor};
+
+      &[disabled] {
+        opacity: 0.8;
+        border-bottom: 1px solid ${colors.neutralBackgroundColor};
+      }
+    }
+  }
+
+  > section {
+    overflow: auto;
+  }
+
+  .captcha-error {
+    text-align: center;
+    margin: 5px;
+    padding: 5px;
+    background: #f5cdcd7a;
+    border-radius: 10px;
+    border: red dotted 1px;
+  }
+
+  .captcha-holder {
+    text-align: center;
+  }
+
+  @media (max-width: 512px), (max-height: 512px) {
+    &.toolbar-iphone-x {
+      input, input:focus {
+        background-color: white;
+      }
+    }
+
+    position: fixed;
+    top: 0;
+    width: 100%;
+    max-height: 100%;
+    right: 0;
+    left: 0;
+    margin: 0;
+    padding-right: max(constant(safe-area-inset-right), 0px);
+    padding-left: max(constant(safe-area-inset-left), 0px);
+    padding-right: max(env(safe-area-inset-right), 0px);
+    padding-left: max(env(safe-area-inset-left), 0px);
+    margin-top: constant(safe-area-inset-top);
+    margin-top: env(safe-area-inset-top);
+    transform: translate3d(0, 0, 0) !important;
+    z-index: 1000000000;
+    border-radius: 0;
+
+    &:not(.is-expanded) {
+      top: 60px;
+      left: 10px;
+      width: calc(100% - 80px);
+      max-height: 100%;
+      max-width: 320px;
+      margin: 0;
+    }
+
+    > header, .search-results, .category-menu {
+      padding: 0
+    }
+
+    .search-results .link-button {
+      margin: 0;
+    }
+
+    @media (max-height: 400px) {
+      .category-button {
+        flex-basis: 16.666666% !important;
+      }
+    }
+  }
+`;
+
+/* TODO: prepare stylable component 
+const contentBelowSearchField = styled.button`
+    overflow-x: hidden;
+    overflow-y: auto;
+    display: flex;
+    flex-direction: column;
+    justify-content: center;
+    
+    padding: 1em;
+    text-align: center;
+  
+    .captcha-content {
+    }
+
+    h3.captcha-help {
+    }
+    
+    small.captcha-explanation {
+
+    }
+  }
+`;
+*/
+
+export default class PhotoUploadCaptchaToolbar extends React.Component<Props, State> {
+  props: Props;
+
+  state: State = {
+    waitingForCaptcha: false,
+    captchaError: null,
+    captcha: null,
+  };
+
+  inputField: ?HTMLInputElement;
+  backLink: ?React.ElementRef<typeof CloseLink>;
+  goButton: ?React.ElementRef<'button'>;
+  checkCaptchaTimer: number | null;
+
+  componentDidMount() {
+    if (!this.props.hidden) {
+      this.startTimer();
+      this.focus(); // Focus input field on start
+    }
+
+    accessibilityCloudImageCache.getCaptcha().then(captcha => {
+      this.setState({waitingForCaptcha: false, captcha, captchaError: false});
+    }).catch(e => {
```

Comment:
A console.log would be cool for this catch-all – if something breaks it makes our lives easier

Reply:
Will take a look